### PR TITLE
fb_storage: support for modern chef and array-skipping

### PR DESCRIPTION
**Pull request by jaymzh**
Depends on #101

* Use new `node.filesystem_data` from #101
* Fix parted-size validator to handle partial fractions
* Handle `_skip` on arrays (incase one has an array controlled by something else)
* Handle ohai for newer Chef versions
* tests for new functionality
* fix existing test
* lots of lint fixes

### CODE REVIEW

File: cookbooks/fb_storage/libraries/format_devices_provider.rb [hunk 1]
```diff
@@ -238,7 +239,7 @@ def get_primary_work(storage)
           partitions.each do |p|
             storage.arrays.each do |array, info|
               if (info['members'].include?(p) ||
-                  (info['journal'] && info['journal'].include?(p))) &&
+                  (info['journal']&.include?(p))) &&
```
Comment (davide125) [inline cookbooks/fb_storage/libraries/format_devices_provider.rb]: @NaomiReeves, will this work fine with our current Chef version?

File: cookbooks/fb_storage/recipes/default.rb [hunk 2]
```diff
@@ -108,7 +110,7 @@
                  'It must be a number with an optional suffix of %%kmgt'
           %w{start end}.each do |disp|
             unless partition["partition_#{disp}"].match(
-              /^\d+([KkMmGgTt\%](iB)?)?$/,
+              /^\d+(\.\d+)?([KkMmGgTt\%](iB)?)?$/,
```
Comment (davide125) [inline cookbooks/fb_storage/recipes/default.rb]: Can you elaborate on this?

Comment (jaymzh) [reply cookbooks/fb_storage/recipes/default.rb]: I did, in the description. It allows for decimals. `6.5G` :)

Comment (jaymzh) [reply cookbooks/fb_storage/recipes/default.rb]: Oh LOL it says "partial fractions" I meant "partial sizes" or "fraction sizes" heh.

File: cookbooks/fb_storage/libraries/format_devices_provider.rb (hunk 1 above)
Comment (jaymzh) [reply cookbooks/fb_storage/libraries/format_devices_provider.rb]: I used it internally a bunch before i left, so I would imagine?

**Pull request closed without merging**